stdlib/io/src/main/ballerina/io/print.bal

```diff
@@ -20,8 +20,17 @@ public extern function print(any... a);
 # Prints an any value to the STDOUT in a new line.
 public extern function println(any... a);
 
-# Returns a formatted string using the specified format string and arguments.
+# Returns a formatted string using the specified format string and arguments. Following format specifiers are allowed.
+#    b              boolean
+#    B              boolean (ALL_CAPS)
+#    d              int
+#    f              float
+#    x              hex
+#    X              HEX (ALL_CAPS)
+#    s              string (This specifier is applicable for any of the supported types in Ballerina.
+#                   These values will be converted to their string representation.)
```

Comment:
would it make sense to allow 'S' too? 's' would have a string representation as is (mix of lower case, upper case), 'S' would be all upper case?

Reply:
Then that means `s` should print all lowercase. It doen't behave like that at the moment, so it does not make sense to have `S`. If someone wants to print a string in uppercase, they can use `toUpper()` function.

Reply:
not necessarily right? For example with Java the behaviour is as follows:
- ```String.format("%s", "Ballerina") //Ballerina```
- ```String.format("%S", "Ballerina") //BALLERINA```

Reply:
Take the below example.

```java
int x = 0xAAAaaa;
String.format("%x", x) // aaaaaa - This should have been 0xAAAaaa to be consistent with `s`.
String.format("%X", x) // AAAAAA
```

So there seem to be inconsistency between `x` and `s` converters. This could have been due to convention of writing hex values in either lowercase or uppercase. But this clearly contradicts with the behavior displayed in s/S.

Maybe Java guys didn't thought it through? Just because it is in java, it does not mean it is the best way or the right way.

Reply:
Since we use %s to give string representation for all supported types in Ballerina, is it a good idea to make everything uppercase? I mean it would uppercase xml/json value if someone tries to print it using %S

Reply:
yeah was just thinking out loud. Given Riyafa's example, agreed that it might not make sense with all the supported types, so +1 for the current implementation.